src/teacher/commands/assign-subject.command.ts

```diff
@@ -75,22 +82,24 @@ export class AssignSubjectCommand {
     teachers: Array<any>,
     body: TeacherAssingDto,
   ) {
-    const subject = await this.prisma.subject.findUnique({
-      where: { id: body.subject_id },
-    });
-
     for (const teacher of teachers) {
-      const email = teacher.user.email;
-      const teacherName = teacher.user.name;
-      const subjectName = `${subject.code} - ${subject.name}`;
-      const subjectId = body.subject_id;
-
-      await this.emailService.sendEmailAssignSubject(
-        email,
-        teacherName,
-        subjectName,
-        subjectId,
-      );
+      for (const subject_id of body.subject_ids) {
+        const subject = await this.prisma.subject.findUnique({
+          where: { id: subject_id },
+        });
```

Comment:
Você já buscou todas as disciplinas na validação que faz no início do comando, então não precisa acessar uma por uma de novo aqui, é só pegar o que foi retornado lá